pkg/distributor/ha_tracker.go

```diff
@@ -371,7 +372,12 @@ func (h *defaultHaTracker) loop(ctx context.Context) error {
 		return true
 	})
 
```

Comment:
[nit] If the code reach this point and the context has not done, then we have an unexpected issue. We could check it and log an error log.

Reply:
I can add something in a separate PR.